woocommerce/wc-admin
# Add Table component

See #92; This is a first-iteration of a table component. I've put a demo inside a card on the test report page:

<img width="1257" alt="screen shot 2018-06-15 at 11 15 41 am" src="https://user-images.githubusercontent.com/541093/41475777-ae0b6aa8-708d-11e8-93a1-ec07d2fa20fb.png">

This component allows for sorting of sortable columns (so the order count is not sortable, since it's an object - a link to those orders). This also toggles to being horizontally scrollable on smaller screens.

<img width="405" alt="screen shot 2018-06-15 at 11 53 51 am" src="https://user-images.githubusercontent.com/541093/41477611-d1f6a0f4-7092-11e8-8c1a-6b87c601e365.png">

I've added a file of "mock data" which is used to populate this table, it matches the data format from the proposed API (see p7bje6-12R-p2). There's more info about how [the table component works in its README.](https://github.com/woocommerce/woo-dash/blob/48cd1b6b6108776b5b9db92ca69022029c4b3fc1/client/components/table/README.md)

**The sorting functionality needs more work.** The money numbers are all converted to strings with `getCurrencyFormatString` so we can show the correct currency symbol/decimal precision. This means we're doing string sorting, and it's not exactly correct (try sorting by the tax col). We also can't sort react objects, so the orders count link is not a sortable col.

I'm wondering if [`getRowsContent `](https://github.com/woocommerce/woo-dash/blob/48cd1b6b6108776b5b9db92ca69022029c4b3fc1/client/analytics/report/index.js#L23) should return a list of objects like `{ label: string/element, value: number }` and each row-item should render `label` but sort using `value`? Does that make `getRowsContent `/the `rows` prop too unwieldy?

**To test**

- View the demo table at `/wp-admin/admin.php?page=woodash#/analytics/revenue`
- Interact with the sorting
- Shrink down your browser size, try scrolling the table
- Check that the readme makes sense

#92 also says it should have a "Table summary row that appears at the bottom of the table and is not really part of the table structure." – I don't think this is part of the table component, so it's not been included here. Maybe we'll end up creating a parent component with a table/summary/pagination combo for use across analytics reports, but I don't think it belongs with the base table.

Note: I know this doesn't match the design exactly, I'm waiting for the library to settle before we set up our color/sizing variables.
